src/library.cpp

```diff
@@ -73,10 +78,21 @@ bool Library::removeBookmark(const std::string& zimId, const std::string& url)
   return false;
 }
 
-
+void Library::removeAllBookmarksByZimID(const std::string& zimId)
+{
+  auto it=m_bookmarks.begin();
+  while (it!=m_bookmarks.end()) {
+    if (it->getBookId() == zimId) {
+      m_bookmarks.erase(it);
+    } else {
+      it++;
+    }
+  } 
+}
```

Comment:
I somehow (always) dislike this way of doing to remove a this from a vector.

I've just discover this idiom : https://en.wikipedia.org/wiki/Erase%E2%80%93remove_idiom

We should use it :)